Update ROS launch documentation (frontend documents) (#247)
ros2/design: articles/151_roslaunch_xml.md

<arg name="output_path" description="Output path for some processing pipeline"/>
```

#### `<executable>` Tag

##### Description

> The `<executable>` tag allows for executing any executable in the form of local OS process.

Reviewer: @ivanpauno consider
```suggestion
The `<executable>` tag allows for executing any executable as a local OS process.
```

### Built-in Substitutions

`$(find-pkg <pkg-name>)`

> : Substituted by the install prefix path of the package.

Reviewer: @ivanpauno consider
```suggestion
: Substituted by the install prefix path of the given package.
```
instead.

  Forward and backwards slashes will be resolved to the local filesystem convention.
  Substitution will fail if the executable cannot be found.

`$(exec-in-package <exec-name> <package-name>)`
: Substituted by the path to the executable in the local filesystem.

>   Lookups at the `lib` folder of the package.

Reviewer: @ivanpauno consider
```suggestion
  Executables are looked up in the `lib` directory of the package.
```
